revv2703/cogniquill
# Feature: Introduce PDF Upload and Rendering

## Description

🚀 **Feature: PDF Upload and Rendering**

This pull request introduces a new feature that enables users to upload PDF files and seamlessly render them within the application. This enhancement enhances our SaaS platform by expanding the range of supported file types and providing users with a more versatile document viewing experience.

## Changes Made

- Implemented PDF file upload functionality using [UploadThing]
- Integrated PDF rendering capabilities to display uploaded PDFs directly in the application.

## Screenshots

![image](https://github.com/revv2703/cogniquill/assets/53593614/dedde5a0-b207-40d1-ad68-38f06f815c18)
![image](https://github.com/revv2703/cogniquill/assets/53593614/1c0c7a3b-e74a-4daa-9432-063d1a5d9948)


## How to Test

1. **Upload a PDF:**
   - Navigate to the [specific section/page] where file uploads are supported.
   - Click on the "Upload" button.
   - Choose a PDF file from your local machine and upload it.

2. **Verify PDF Rendering:**
   - After the upload is complete, navigate to the [relevant section/page] where rendered PDFs are displayed.
   - Confirm that the uploaded PDF is rendered and viewable within the application.

## Dependencies

- Uploadthing
- react-dropzone
- react-toast
- react-dialog
- react-progress

## Related Issues

None

## Checklist

- [x] Code has been tested locally and works as expected.
- [x] Documentation has been updated to reflect the new feature.

## Additional Notes

[Include any additional information or notes that might be helpful for reviewers.]
